RHDEVDOCS-3308 [Recurring] Release Notes, Known Issues, and Bug Fixes… (#37369)
openshift/openshift-docs: modules/op-release-notes-1-6.adoc

- Support for all clustertask/task/taskrun/pipeline/pipelinerun cmd with v1alpha1 resources has been deprecated
and will be removed after two releases.

=== Tekton Triggers v0.16.0

* Remove Redundant Status Tag from EventListener Metrics [#1166](https://github.com/tektoncd/triggers/pull/1166)

Breaking Change:

Status label has been removed from metric eventlistener_http_duration_seconds_*. Please remove querying base on status label.

=== Tekton Triggers v0.15.0

* Send EL response before Triggers finish processing [#1132](https://github.com/tektoncd/triggers/pull/1132)

Breaking Change:

With this change the EventListener will stop responding with 201 Created status code when it creates resources. Instead it will always respond with a 202 Accepted response code.

* Removes podTemplate field from EventListener [#1118](https://github.com/tektoncd/triggers/pull/1118)

Breaking Change:

The podTemplate filed which was deprecated as part of #1100 has now been removed.

* Removes the replicas field [#1113](https://github.com/tektoncd/triggers/pull/1113)

Breaking Change:

The deprecated replicas field has now been removed.

[id="known-issues-1-6_{context}"]
== Known issues

> 

Author: Updated. Thanks.

> 

Author: @SNiemann15  Do you happen to have the Jira/BZ/GitHub issue numbers for these items?

[id="fixed-issues-1-6_{context}"]
== Fixed issues

> 

Author: @SNiemann15 Do you happen to have the Jira/BZ/GitHub issue numbers for this item?

* With this update, when you use the `tkn pipelinerun delete --all` command, you can use the new `--label <string>` option to filter the pipeline runs by label. Optionally, you can use the `--label` option with `=` and `==` as equality operators, or `!=` working as an inequality operator. For example, the `tkn pipelinerun delete --all --label asdf` and  `tkn pipelinerun delete --all --label==asdf` commands both delete all the pipeline runs that have the `asdf` label. (link:https://github.com/tektoncd/cli/pull/1402[#1402])

asdflkasdflkj

* Get tekton components version from configmap [#1393](https://github.com/tektoncd/cli/pull/1393)

=== Tekton Triggers v0.16.0

* Triggers support feature-flags ConfigMap to configure feature flags [#1110](https://github.com/tektoncd/triggers/pull/1110)

* Add Event Count for EventListener Metrics [#1160](https://github.com/tektoncd/triggers/pull/1160). Added a new counter Metric `eventlistener_event_count` for requests received at EventListener sink.

* Add config-defaults configmap to set default values [#1182](https://github.com/tektoncd/triggers/pull/1182)

> * Triggers now support config-defaults configmap to set default values and as of now serviceAccountName added to config-defaults.

Author: Updated. Thanks.

=== Tekton Triggers v0.15.0

* Add v1beta1 Go API types [#1103](https://github.com/tektoncd/triggers/pull/1103)

> Triggers now supports a new apiVersion - v1beta1. The v1beta1 apiVersion will follow the beta deprecation policy i.e. features will have a 9 month deprecation period.

Author: Updated. Thanks.

// Module included in the following assembly:
//
// * cicd/pipelines/op-release-notes.adoc

[id="op-release-notes-1-6_{context}"]
= Release notes for {pipelines-title} General Availability 1.6

With this update, {pipelines-title} General Availability (GA) 1.6 is available on {product-title} 4.9.

[id="new-features-1-6_{context}"]
== New features

In addition to the fixes and stability improvements, the following sections highlight what is new in {pipelines-title} 1.6.

[id="new-features-cli-0-21-0-release-1-6_{context}"]
=== CLI 0.21.0 release

> * With this update, you can configure a pipeline or task `start` command to return a YAML or JSON-formatted string by using the `--output <string>`, where `<string>` is `yaml` or `json`. Otherwise, without the `--output` option, the `start` command returns a human-friendly message that is hard for other programs to parse. Returning a YAML or JSON-formatted string is useful for continuous integration (CI) environments.  For example, after a resource is created, you can use `yq` or `jq` to parse the YAML or JSON-formatted message about the resource and wait until that resource is terminated without using the `showlog` option. (link:https://github.com/tektoncd/cli/pull/1326[#1326])

Author: Thanks. Updated.

* This update adds support for the triggers resources to work with `v1beta1` resources (link:https://github.com/tektoncd/cli/pull/1446[#1446], link:https://github.com/tektoncd/cli/pull/1449[#1449], link:https://github.com/tektoncd/cli/pull/1450[#1450], link:https://github.com/tektoncd/cli/pull/1454[#1454], link:https://github.com/tektoncd/cli/pull/1455[#1455])

* This update adds an `ignore-running` option to the `tkn pipelinerun delete` and `tkn taskrun delete` commands. (link:https://github.com/tektoncd/cli/pull/1445[#1445])

* This update adds a `create` subcommand to the `tkn task` and `tkn clustertask` commands. (link:https://github.com/tektoncd/cli/pull/1359[#1359])

[id="new-features-cli-0-20-0-release-1-6_{context}"]
=== CLI 0.20.0 release

> * With this update, if you use the `tkn pipelinerun delete --all` command, you can preserve runs that are younger than a specified number of minutes by using the new `--keep-since <minutes>` option. (link:https://github.com/tektoncd/cli/pull/1392[#1392])

Reviewer: Basically the same thing is already documented above.

Author: Updated throughout. Thanks.

+
With the current release, the `v1alpha1` features are deprecated and are candidates for removal after nine months. Begin using `v1beta1` features instead. (link:https://github.com/tektoncd/triggers/pull/1103[#1103])

[id="new-features-pipelines-operator-1-6_{context}"]
=== {pipelines-title} Operator

* The current release enables you to configure node selection by specifying the `nodeSelector` and `toleration` values in the `TektonConfig` custom resource (CR). The Operator adds these values to all deployments it creates.
+
To configure node selection for the Operator's controller and webhook deployment, you edit the `config.nodeSelector` and `config.toleration` fields in the specification for the `Subscription` CR after installing the Operator.
+
To deploy on an infra node, you update the `TektonConfig` and `Subscription` CRs with the infra label as `nodeSelector` and `toleration` so that these labels are applied to the infra node. (link:https://issues.redhat.com/browse/SRVKP-1544[SRVKP-1544])

[id="deprecated-features-1-6_{context}"]
== Deprecated features

[id="deprecated-cli-0-21-0-release-1-6_{context}"]
=== CLI 0.21.0 release

> * In CLI 0.21.0, support for all `v1alpha1` resources for `clustertask`, `task`, `taskrun`, `pipeline`, and `pipelinerun` commands are deprecated. These resources will be removed after two releases.

Reviewer: `These resources will be removed after two releases.`  - does this refer to upstream releases or product releases? In previous releases we only stated something like `... are now deprecated and will be removed in a future release.`

Author: Updated throughout. Thanks.

[id="deprecated-tekton-0-16-0-1-6_{context}"]
=== Tekton Triggers v0.16.0

* In Tekton Triggers v0.16.0, the redundant `status` label is removed from the metrics for the `EventListener` resource. (link:https://github.com/tektoncd/triggers/pull/1166[#1166])
+
[IMPORTANT]
====
Breaking change: The `status` label has been removed from the `eventlistener_http_duration_seconds_*` metric.
Remove queries that are based on the `status` label.
====

[id="deprecated-tekton-0-15-0-1-6_{context}"]
=== Tekton Triggers v0.15.0

> * With the current release, the `v1alpha1` features are deprecated and are candidates for removal after nine months. With this update, you can begin using the `v1beta1` Go API types instead. Triggers now supports the `v1beta1` API version. (link:https://github.com/tektoncd/triggers/pull/1103[#1103])

Reviewer: `candidates for removal after nine months` - the same as above

Author: Updated throughout. Thanks.

+
[IMPORTANT]
====
Breaking change: The deprecated `replicas` field has been removed.
====

[id="deprecated-features-pipelines-operator-1-6_{context}"]
=== {pipelines-title} Operator

* In {pipelines-title} 1.6, the values of `HOME="/tekton/home"` and `workingDir="/workspace"` are removed from the `Steps` specification.
+
Instead, {pipelines-title} sets `HOME` and `workingDir` to the values defined by your `Steps` containers. You can override these values in your `Steps` specification.
+
To use the older behavior, you can change the `disable-working-directory-overwrite` and `disable-home-env-overwrite` fields in the `TektonConfig` CR to `false`:
+
[source,yaml]
----
apiVersion: operator.tekton.dev/v1alpha1
  kind: TektonConfig
  metadata:
    name: config
  spec:
    pipeline:
      disable-working-directory-overwrite: false
      disable-home-env-overwrite: false
  ...
----
+
[IMPORTANT]
====
The `disable-working-directory-overwrite` and `disable-home-env-overwrite` fields in the `TektonConfig` CR  are deprecated and are candidates for removal after nine months.
====
+

> (link:https://issues.redhat.com/browse/SRVKP-1465[SRVKP-1465])

Reviewer: Let's remove this link even though it points to our own issue tracker

Author: Updated. Thanks.

[id="known-issues-1-6_{context}"]
== Known issues

* When you run Maven and Jib-Maven cluster tasks, the default container image is supported only on Intel (x86) architecture. Therefore, tasks will fail on IBM Power Systems (ppc64le), IBM Z, and LinuxONE (s390x) clusters. As a workaround, you can specify a custom image by setting the `MAVEN_IMAGE` parameter value to `maven:3.6.3-adoptopenjdk-11`.
// issue # is unknown.

* On IBM Power Systems, IBM Z, and LinuxONE, the `s2i-dotnet` cluster task is unsupported.
// issue # is unknown.

> * Before you install `tekton` catalog tasks on IBM Power Systems (ppc64le), IBM Z, and LinuxONE(s390x) using `tkn hub`, verify if the task can be executed on these platforms. To check if `ppc64le` and `s390x` are listed in the "Platforms" section of the task information, you can run the following command: `tkn hub info task <name>`

Reviewer: `tekton` is usually displayed as ordinary text

Author: Updated. Thanks.

// Module included in the following assembly:
//
// * cicd/pipelines/op-release-notes.adoc

[id="op-release-notes-1-6_{context}"]
= Release notes for {pipelines-title} General Availability 1.6

> With this update, {pipelines-title} General Availability (GA) 1.6 is available on {product-title} 4.7, 4.8 and 4.9.

Reviewer 2: 1.6 is only going to be available on 4.9 though
cc @siamaksade

Author: Updated. Thanks.

[id="compatibility-support-matrix-1-6_{context}"]
== Compatibility and support matrix

> Some features in this release are currently in Technology Preview. These experimental features are not intended for production use.
> 
> link:https://access.redhat.com/support/offerings/techpreview[Technology Preview Features Support Scope]

Reviewer 3: ```suggestion
Some features in this release are currently in Technology Preview. These experimental features are not intended for production use.

link:https://access.redhat.com/support/offerings/techpreview[Technology Preview Features Support Scope]
```
```suggestion
Some features in this release are currently in link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]. These experimental features are not intended for production use.
```
Suggestion to add the link to text

Author: Updated. Thanks.

In the table, features are marked with the following statuses:

- *TP*: _Technology Preview_

> - *GA*: _General Availability_

Reviewer 3: Maybe use a definition list here? Just a suggestion

Author: Updated. Thanks.

Note the following scope of support on the Red Hat Customer Portal for these features:

.Compatibility and support matrix
[cols="1,1,1",options="header"]
|===
| Feature   | Version       | Support Status
| {product-title} | 4.9     | GA
| Pipelines Operator | 1.6  | GA
| Pipelines | 0.28.x        | GA
| CLI       | 0.21.x        | GA
| Catalog   | 0.28          | GA
| Triggers  | 0.16.x        | GA
| Pipeline resources | -    | TP
|===

For questions and feedback, you can send an email to the product team at pipelines-interest@redhat.com.

[id="new-features-1-6_{context}"]
== New features

In addition to the fixes and stability improvements, the following sections highlight what is new in {pipelines-title} 1.6.

> === CLI 0.21.0 release

Reviewer 3: do all headings need an ID?

Author: Updated. Thanks. I have removed the versions but kept the categories that were presented to me in this source document: https://docs.jboss.org/display/RHDEVDOCS/OpenShift+Pipelines+1.6+documentation+plan#OpenShiftPipelines1.6documentationplan-Candidatesfor1.6RNs

* With this update, you can configure a pipeline or task `start` command to return a YAML or JSON-formatted string by using the `--output <string>`, where `<string>` is `yaml` or `json`. Otherwise, without the `--output` option, the `start` command returns a human-friendly message that is hard for other programs to parse. Returning a YAML or JSON-formatted string is useful for continuous integration (CI) environments.  For example, after a resource is created, you can use `yq` or `jq` to parse the YAML or JSON-formatted message about the resource and wait until that resource is terminated without using the `showlog` option. (link:https://github.com/tektoncd/cli/pull/1326[#1326])

* With this update, you can authenticate to a registry with Podman's `auth.json` authentication file. For example, you can use `tkn bundle push` to push to a remote registry using podman instead of docker CLI. (link:https://github.com/tektoncd/cli/pull/1430[#1430])

> * With this update, if you use the `tkn taskrun delete --all` command, you can preserve runs that are younger than a specified number of minutes by using the new `--keep-since <minutes>` option. For example, to keep task runs that are less than five minutes old, you enter `$ tkn taskrun delete -all --keep-since 5`. (link:https://github.com/tektoncd/cli/pull/1435[#1435])

Reviewer 3: Consider maybe laying this out so that it's easier to copy the commands?
e.g.
```
* With this update, you can use the `--keep-since <minutes>` flag with the `tkn taskrun delete --all` command to preserve runs that are younger than a specified number of minutes (link:https://github.com/tektoncd/cli/pull/1435[#1435]).
+
.Example command to keep task runs that are less than five minutes old
[source,terminal]
----
$ tkn taskrun delete -all --keep-since 5
----

Author: Updated. Thanks.

> * With this update, when you delete task runs or pipeline runs, you can use the `--parent-resource` and `--keep-since` options together. For example, the `tkn pipelinerun delete --pipeline pipelinename --keep-since 5` command preserves pipeline runs whose parent resource is named `pipelinename` and whose age is five minutes or less. The `$ tkn tr delete -t taskname --keep-since 5` and `$ tkn tr delete --clustertask taskname --keep-since 5` commands work similarly for task runs. (link:https://github.com/tektoncd/cli/pull/1443[#1443])

Reviewer 3: ```suggestion
* With this update, when you delete task runs or pipeline runs, you can use the `--parent-resource` and `--keep-since` options together. For example, the `tkn pipelinerun delete --pipeline pipelinename --keep-since 5` command preserves pipeline runs whose parent resource is named `pipelinename` and whose age is five minutes or less. The `tkn tr delete -t taskname --keep-since 5` and `tkn tr delete --clustertask taskname --keep-since 5` commands work similarly for task runs. (link:https://github.com/tektoncd/cli/pull/1443[#1443])
```

> * This update adds support for triggers resources to work with `v1beta1` resources (link:https://github.com/tektoncd/cli/pull/1446[#1446], link:https://github.com/tektoncd/cli/pull/1449[#1449], link:https://github.com/tektoncd/cli/pull/1450[#1450], link:https://github.com/tektoncd/cli/pull/1454[#1454], link:https://github.com/tektoncd/cli/pull/1455[#1455])

Reviewer 3: ```suggestion
* This update adds support for trigger resources to work with `v1beta1` resources (link:https://github.com/tektoncd/cli/pull/1446[#1446], link:https://github.com/tektoncd/cli/pull/1449[#1449], link:https://github.com/tektoncd/cli/pull/1450[#1450], link:https://github.com/tektoncd/cli/pull/1454[#1454], link:https://github.com/tektoncd/cli/pull/1455[#1455])
```
suggestion if it doesn't change the meaning

Reviewer 4: `triggers` is a project repo name https://github.com/tektoncd/triggers so i would suggest lets keep as it is

Author: Left it as is. Thanks.

* This update adds an `ignore-running` flag to the `tkn pipelinerun delete` and `tkn taskrun delete` commands. (link:https://github.com/tektoncd/cli/pull/1445[#1445])

* This update adds a `create` subcommand to the `tkn task` and `tkn clustertask` commands. (link:https://github.com/tektoncd/cli/pull/1359[#1359])

> === CLI 0.20.0 release

Reviewer 3: Do all headings need IDs?

Reviewer 3: also seems like it contains some of the same things as _CLI 0.21.0 release_?

Author: Updated. Thanks.

* With this update, if you use the `tkn pipelinerun delete --all` command, you can preserve runs that are younger than a specified number of minutes by using the new `--keep-since <minutes>` option. For example, to keep pipeline runs that are less than five minutes old, you enter `$ tkn pipelinerun delete -all --keep-since 5`. (link:https://github.com/tektoncd/cli/pull/1392[#1392])

* With this update, when you use the `tkn pipelinerun delete --all` command, you can use the new `--label <string>` option to filter the pipeline runs by label. Optionally, you can use the `--label` option with `=` and `==` as equality operators, or `!=` as an inequality operator. For example, the `tkn pipelinerun delete --all --label asdf` and  `tkn pipelinerun delete --all --label==asdf` commands both delete all the pipeline runs that have the `asdf` label. (link:https://github.com/tektoncd/cli/pull/1402[#1402])

> * With this update, you can fetch the version of installed Tekton components from the `ConfigMap` or, if the `ConfigMap` is not present, from the deployment controller. (link:https://github.com/tektoncd/cli/pull/1393[#1393])

Reviewer 3: use "config map" per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

Author: Updated. Thanks.

> === Tekton Triggers v0.16.0

Reviewer 3: Do all headings need IDs?

Author: Updated. Thanks.

> * With this update, triggers support the `configmap feature-flags` and `configmap config-defaults` commands to configure feature flags and to set default values, and adds `serviceAccountName` to the defaults. (link:https://github.com/tektoncd/cli/pull/1110[#1110], link:https://github.com/tektoncd/cli/pull/1182[#1182])

Reviewer 3: ```suggestion
* With this update, triggers support the `oc configmap feature-flags` and `oc configmap config-defaults` commands to configure feature flags and to set default values, and adds `serviceAccountName` to the defaults. (link:https://github.com/tektoncd/cli/pull/1110[#1110], link:https://github.com/tektoncd/cli/pull/1182[#1182])
```
seems like an incomplete command, should they have `oc` or `tkn` before `configmap`?
Also, the "and adds `serviceAccountName` to the defaults." bit seems like it could be its own bullet, especially if they have two separate github issues / PRs?

Reviewer 4: * With this update, triggers support the `feature-flags` and `config-defaults` configmap to configure feature flags and to set default values respectively. (link:https://github.com/tektoncd/cli/pull/1110[#1110], link:https://github.com/tektoncd/cli/pull/1182[#1182])

How about this?

Author: Updated. Thanks.

Reviewer 4: We need to change links 
Its pointing to `cli`

correct links
https://github.com/tektoncd/triggers/pull/1182
https://github.com/tektoncd/triggers/pull/1110

Reviewer 3: @savitaashture makes sense to me

Author: Updated. Thanks.

* This update adds a new metric, `eventlistener_event_count`, that you can use to count events received by the `EventListener` sink. (link:https://github.com/tektoncd/triggers/pull/1160[#1160])

> === Tekton Triggers v0.15.0

Reviewer 3: ID?

Author: Updated. Thanks.

* This update adds `v1beta1` Go API types. (link:https://github.com/tektoncd/triggers/pull/1103[#1103])
+

> With this update, Triggers now supports a new `apiVersion` value, `v1beta1`. The `v1alpha1` version will follow the beta deprecation policy: `v1alpha1` features will be deprecated over a 9-month period.

Reviewer 3: ```suggestion
With this update, triggers now support the `v1beta1` API version. The `v1alpha1` version will follow the beta deprecation policy: `v1alpha1` features will be deprecated over a 9-month period.
```
suggestion for readability; I think it's safe to assume users would know "API version" refers to the `apiVersion` value in the YAML, but its up to you

Author: Updated. Thanks.

> 

Author: In which section should I place the following release note? If you look at [the TOC in the preview](https://deploy-preview-37369--osdocs.netlify.app/openshift-enterprise/latest/cicd/pipelines/op-release-notes), I've placed it at the same level as:

- CLI 0.21.0 release
- CLI 0.20.0 release
- Tekton Triggers v0.16.0
- Tekton Triggers v0.15.0

However, it seems like I should place it _**within**_ one of those sections, or something similar. I couldn't determine this from the context. What do you think?

Reviewer 2: Not sure I follow the question 🤔

Reviewer 2: Oh, I think I understand. Configuring Node selection is a feature of the Operator (so OpenShift pipelines "installation")

Author: Thanks @vdemeester. I created the following `=== {pipelines-title} Operator` section and placed that item as a release note within it. How does that look to you?

Author: @vdemeester, I created the following `=== {pipelines-title} Operator` section to contain the release note. WDYT?

[id="configuring-node-selection-1-6_{context}"]
=== Configuring Node Selection
// TBD - Where does this release note belong?

> * The current release enables you to configure node selection by specifying the `nodeSelector` and `toleration` values in the `TektonConfig` custom resource (CR) YAML. The Operator adds these values to all deployments it creates.

Reviewer 3: ```suggestion
* The current release enables you to configure node selection by specifying the `nodeSelector` and `toleration` values in the `TektonConfig` custom resource (CR). The Operator adds these values to all deployments it creates.
```
probably don't need YAML here, particularly if this can be done using `tkn` or ODC as well, without YAML?

Reviewer 4: I believe this we can done by YAML i think
/cc @sm43

Author: Thank you, both. I'm not sure what to make of this feedback - I'll go ahead and remove `YAML`. Let me know if further changes are needed.

+

> To configure node selection for the Operator's controller and webhook deployment, you edit the `Subscription` CR after installing the Operator.
> The `Subscription` CR has the following fields in its specification:
> - `config.nodeSelector`
> - `config.toleration`

Reviewer 3: ```suggestion
To configure node selection for the Operator's controller and webhook deployment, edit the `Subscription` CR after installing the Operator. The `Subscription` CR has the following fields in its specification:

- `config.nodeSelector`
- `config.toleration`
```

+
To deploy on an infra node, you update the `TektonConfig` and `Subscription` CRs with the infra label as `nodeSelector` and `toleration` so that these labels are applied to the infra node. (link:https://issues.redhat.com/browse/SRVKP-1544[SRVKP-1544])

[id="deprecated-features-1-6_{context}"]
== Deprecated features

=== CLI 0.21.0 release

* In CLI 0.21.0, support for all `v1alpha1` resources for `clustertask`, `task`, `taskrun`, `pipeline`, and `pipelinerun` commands are deprecated. These resources will be removed after two releases.

=== Tekton Triggers v0.16.0

* In Tekton Triggers v0.16.0, the redundant status tag is removed from the `EventListener` metrics. (link:https://github.com/tektoncd/triggers/pull/1166[#1166])
+
[IMPORTANT]
====

> Breaking change: The Status label has been removed from the `eventlistener_http_duration_seconds_*` metric.

Reviewer 3: ```suggestion
Breaking change: The status label has been removed from the `eventlistener_http_duration_seconds_*` metric.
```
make consistent?

Author: Updated. Thanks.

> Remove queryies that are based on the status label.

Reviewer 3: ```suggestion
Remove queries that are based on the status label.
```
not sure if this is a typo?

Author: Updated. Thanks.

====

=== Tekton Triggers v0.15.0

> * With the current release, the `EventListener` sends a response before Triggers finishes processing. (link:https://github.com/tektoncd/triggers/pull/1132[#1132])

Reviewer 3: I think "triggers" should be lowercase throughout?

+
[IMPORTANT]
====
Breaking change: With this change, the `EventListener` stops responding with a `201 Created` status code when it creates resources. Instead, it responds with a `202 Accepted` response code.
====

> * The current release removes the `podTemplate` field from the `EventListener`. (link:https://github.com/tektoncd/triggers/pull/1118[#1118])

Reviewer 3: "specification" below?

> * The current release removes the `podTemplate` field from the `EventListener`. (link:https://github.com/tektoncd/triggers/pull/1118[#1118])

Author: Updated. Thanks.

> * The current release removes the `podTemplate` field from the `EventListener`. (link:https://github.com/tektoncd/triggers/pull/1118[#1118])

Author: `EventListener` resource seems to be the most common usage.

+
[IMPORTANT]
====
Breaking change: The `podTemplate` field, which was deprecated as part of link:https://github.com/tektoncd/triggers/pull/1100[#1100], has been removed.
====

* The current release removes the deprecated `replicas` field from the `EventListener` specification. (link:https://github.com/tektoncd/triggers/pull/1113[#1113])
+
[IMPORTANT]
====
Breaking change: The deprecated `replicas` field has been removed.
====

> 

Author: Same question for this release note - where to place it?

Reviewer 2: This is a deprecation/removal from the Pipeline component of OpenShift Pipelines.

Author: Updated. Thanks.

[id="removing-home-workingdir-values-1-6_{context}"]
=== Removing the HOME="/tekton/home" and workingDir="/workspace" values from Steps
// TBD - Where does this release note belong?

In {pipelines-title} 1.6, the values of `HOME="/tekton/home"` and `workingDir="/workspace"` are removed from the `Steps` specification.

> Instead, {pipelines-title} sets `HOME` and `workingDir` to the values defined by your Steps containers. You can override these values in your `Steps` YAML file.

Reviewer 3: consistently format Steps / `Steps` - maybe steps / `Steps` specification as per other objects? Is `Steps` a kind of CR?

Author: Updated. Thanks.

> To get the older behavior, you can change the following fields in `TektonConfig` to `false`. Their default values are `true`.
> 
> [source,yaml]
> ----
> spec:
>   pipeline:
>     disable-working-directory-overwrite: false
>     disable-home-env-overwrite: false
> ----

Reviewer 3: Can you maybe add part of the metadata showing what `TektonConfig` is?

```suggestion
To use the older behavior, you can change the `disable-working-directory-overwrite` and `disable-home-env-overwrite` fields in the `TektonConfig` CR to `false`:

[source,yaml]
----
apiVersion: operator.tekton.dev/v1alpha1
  kind: TektonConfig
  metadata:
    name: config
  spec:
    pipeline:
      disable-working-directory-overwrite: false
      disable-home-env-overwrite: false
...
----
(https://tekton.dev/docs/operator/tektonconfig/#tekton-config)
```

Author: Thanks! Updated.

[IMPORTANT]
====

> The feature flags are temporarily available. These flags are candidates for complete removal from a future minor release of the upstream product at least nine months from when this release goes live.

Reviewer 3: which feature flags is this referring to?

Author: @vdemeester It looks like Nikhil's geo is in India, so he might be on PTO/holiday. Any chance you could give this a try?

Author: Is this a deprecation notice for the `feature-flags` option. If so, we should give it a deprecation notice similar to this one:
```
With the current release, the `v1alpha1` features are deprecated and are candidates for removal after nine months. Begin using the `v1beta1` features instead. (link:https://github.com/tektoncd/triggers/pull/1103[#1103])
```

Reviewer 2: Yeah I think the kind of make sense. This would be something like "the disable-working-directory-overwrite and disable-home-env-overwrite are still available but are candidate for removal, …" or something similar yes 👍🏼

Author: Updated, @vdemeester. Please review and approve this PR for merge.

====

(link:https://issues.redhat.com/browse/SRVKP-1465[SRVKP-1465])

[id="known-issues-1-6_{context}"]
== Known issues

> * When you run Maven and Jib-Maven cluster tasks on IBM Power Systems (ppc64le), IBM Z, and LinuxONE (s390x) clusters, set the `MAVEN_IMAGE` parameter value to `maven:3.6.3-adoptopenjdk-11`.

Reviewer 3: explain why?

Author: @SNiemann15 Can we provide an explanation why the user should do this?

Reviewer 5: @rolfedh Yes the reason is that the default container can't be used. How about: 

 * When you run Maven and Jib-Maven cluster tasks, the default container image is supported only on Intel (x86) architecture. Therefore, tasks will fail on IBM Power Systems (ppc64le), IBM Z, and LinuxONE (s390x) clusters. 
 As a workaround, you can specify a custom image by setting the `MAVEN_IMAGE` parameter value to `maven:3.6.3-adoptopenjdk-11`.

Author: Thanks! Updated.

// issue # is unknown.

* On IBM Power Systems, IBM Z, and LinuxONE, the `s2i-dotnet` cluster task is unsupported.
// issue # is unknown.

> * Before you install catalog tasks on IBM Power Systems, IBM Z, and LinuxONE using `tkn hub`, make sure to check task information about these platforms. For example, using the following command: `tkn hub info task <name>`

Reviewer 3: what info are we looking for here? What does it indicate?

Author: @SNiemann15 Can we explain this a bit more?

Reviewer 5: @rolfedh I got feedback from Yulia our SME that this is needed because some tasks don't run on P/Z. Yulia also sent me an example output but it's pretty large not sure if you want to have that in the Release Notes. 
Is it ok like this or do you think we need more information? 

* Before you install `tekton` catalog tasks on IBM Power Systems (ppc64le), IBM Z, and LinuxONE(s390x) using `tkn hub`, verify if the task can be executed on these platforms.
To check if `ppc64le` and `s390x` are listed in the "Platforms" section of the task information, you can run the following command: `tkn hub info task <name>`

Author: Updated. Thanks.

// issue # is unknown.

[id="fixed-issues-1-6_{context}"]
== Fixed issues

> The `tkn hub` command is now supported on IBM Power Systems, IBM Z, and LinuxONE.

Reviewer 3: ```suggestion
* The `tkn hub` command is now supported on IBM Power Systems, IBM Z, and LinuxONE.
```

Reviewer 3: no issue #?

Author: Added the bullet, thanks. I don't think issue numbers are available for these IBM-related release notes, right @SNiemann15?

// issue # is unknown.

> === CLI 0.21.0 release

Reviewer 3: IDs for all headings

Author: Updated. Thanks.

* Before this update, the terminal was not available after the user ran a `tkn` command, and the pipeline run was done, even if `retries` were specified. Specifying a timeout in the task run or pipeline run had no effect. This update fixes the issue so that the terminal is available after running the command. (link:https://github.com/tektoncd/cli/issues/1459[#1459])

=== CLI 0.20.0 release

> * Before this update, using the `tkn version --component=<component>` command did not return the component version. This update fixes the issue so that you can get the component version by using the `--component` option. (https://github.com/tektoncd/cli/pull/1408[#1408])

Reviewer 3: ```suggestion
* Before this update, using the `tkn version --component=<component>` command did not return the component version. This update fixes the issue so that you can get the component version. (https://github.com/tektoncd/cli/pull/1408[#1408])
```
I think it's unnecessary to mention the name of the flag again

Author: Updated. Thanks.

* Before this update, when you used the `tkn pr logs` command, it displayed the pipelines output logs in the wrong task order. This update resolves the issue so that logs of completed `PipelineRuns` are listed in the appropriate `TaskRun` execution order. (link:https://github.com/tektoncd/cli/pull/1385[#1385])

> 

Author: Same question for this release note - where to place it?

Reviewer 2: This is also for the Pipeline component (the core) of OpenShift Pipelines.

Author: Updated. Thanks.

[id="pipeline-run-still-executing-1-6_{context}"]
=== Pipeline does not stop running after pipeline changed
// TBD - Where does this release note belong?

Before this update, editing the specification of a running pipeline might prevent the pipeline run from stopping when it was complete. This update fixes the issue by fetching the definition only once and then using the specification stored in the status as the source of truth. This change reduces the probability of a race condition when a `PipelineRun` or a `TaskRun` refers to a `Pipeline` or `Task` that changes while it is running.

> (link:https://issues.redhat.com/browse/SRVKP-718[SRVKP-718])

Author: @abrennan89 I was finding out which component it belonged to. Got that sorted out now. Thanks.

// Module included in the following assembly:
//
// * cicd/pipelines/op-release-notes.adoc

[id="op-release-notes-1-6_{context}"]
= Release notes for {pipelines-title} General Availability 1.6

> With this update, {pipelines-title} General Availability (GA) 1.6 is available on {product-title} 4.9.

Reviewer 6: The module level table is missing, did we decide against it?

Author: I applied your suggestion to remove it.

====
// (link:https://issues.redhat.com/browse/SRVKP-1465[SRVKP-1465])

[id="known-issues-1-6_{context}"]
== Known issues

* When you run Maven and Jib-Maven cluster tasks, the default container image is supported only on Intel (x86) architecture. Therefore, tasks will fail on IBM Power Systems (ppc64le), IBM Z, and LinuxONE (s390x) clusters. As a workaround, you can specify a custom image by setting the `MAVEN_IMAGE` parameter value to `maven:3.6.3-adoptopenjdk-11`.
// issue # is unknown.

* On IBM Power Systems, IBM Z, and LinuxONE, the `s2i-dotnet` cluster task is unsupported.
// issue # is unknown.

* Before you install tasks based on Tekton Catalog on IBM Power Systems (ppc64le), IBM Z, and LinuxONE(s390x) using `tkn hub`, verify if the task can be executed on these platforms. To check if `ppc64le` and `s390x` are listed in the "Platforms" section of the task information, you can run the following command: `tkn hub info task <name>`
// issue # is unknown.

* You cannot use the `nodejs:14-ubi8-minimal` image stream because doing so generates the following errors:
+

> [source,terminal]

Author: Including error messages like this helps users find the release note when they search for the error message.

> [source,terminal]

Author: @ppitonak

> [source,terminal]

Author: @ppitonak This helps the user find the release note when they search for the error message.
